dali/pipeline/workspace/workspace.h

```diff
@@ -628,13 +679,29 @@ class WorkspaceBase : public ArgumentWorkspace {
     return index_map[idx];
   }
 
+  AccessOrder output_order_ = AccessOrder::host();
+  ThreadPool *thread_pool_ = nullptr;
+  cudaEvent_t event_ = nullptr;
+  SmallVector<cudaEvent_t, 4> parent_events_;
+};
 
-  /**
-   * @brief Returns CUDA stream or nullptr, if the stream is unavailable
-   */
-  virtual cudaStream_t stream_impl() const = 0;
+class Workspace : public WorkspaceBase<TensorList> {};
+
+class SampleWorkspace;
+
+template <typename Backend>
+struct LegacyWorkspace {
+  using type = Workspace;
+};
+
+template <>
+struct LegacyWorkspace<CPUBackend> {
+  using type = SampleWorkspace;
 };
 
+template <typename Backend>
+using legacy_workspace_t = typename LegacyWorkspace<Backend>::type;
```

Comment:
This is a helper for the old-style operators that still use `SampleWorkspace` for CPU backend, but otherwise are templated against `Backend`.